src/psyclone/psyir/backend/sir.py

```diff
@@ -345,7 +359,15 @@ def reference_node(self, node):
             raise VisitorError(
                 "Method reference_node in class SIRWriter: SIR Reference "
                 "node is not expected to have any children.")
-        return "{0}make_var_access_expr(\"{1}\")".format(self._nindent,
+        # _scalar_names is a set so duplicates will be ignored. It
+        # captures all unique scalar names as scalars are currently
+        # treated as temporaries (#521 captures this). The simplest
+        # way to declare a scalar temporary in Dawn is to treat it as
+        # a field temporary (as the Dawn backend works out if a scalar
+        # is required).
+        self._scalar_names.add(node.name)
+
+        return "{0}make_field_access_expr(\"{1}\")".format(self._nindent,
```

Comment:
pylint complains about the indentation of the next line.

Reply:
Oops. Fixed.